File: packages/ruby-sdk/lib/strait/operations/environments.rb
# frozen_string_literal: true

module Strait
  module Operations
    # Service for environment CRUD and variable listing.
    class EnvironmentsService < BaseService
      # List all environments.
      # GET /v1/environments
      def list(query: nil)
        _request(:get, "/v1/environments", query: query)
      end

      # Create a new environment.
      # POST /v1/environments
      def create(body)
        _request(:post, "/v1/environments", body: body)
      end

      # Get an environment by ID.
      # GET /v1/environments/{envID}
      def get(env_id)
        _request(:get, "/v1/environments/{envID}", path_params: { "envID" => env_id })
      end

      # Update an environment.
      # PATCH /v1/environments/{envID}
      def update(env_id, body)
        _request(:patch, "/v1/environments/{envID}", path_params: { "envID" => env_id }, body: body)
      end

      # Delete an environment.
      # DELETE /v1/environments/{envID}
      def delete(env_id)
        _request(:delete, "/v1/environments/{envID}", path_params: { "envID" => env_id })
        nil
      end

      # List variables for an environment.
      # GET /v1/environments/{envID}/variables
      def list_variables(env_id)
        _request(:get, "/v1/environments/{envID}/variables", path_params: { "envID" => env_id })
      end
    end
  end
end

File: packages/python-sdk/strait/operations/environments.py
"""Environment management operations."""

from __future__ import annotations

from typing import Any

from strait.operations._base import AsyncBaseService, BaseService


class EnvironmentsService(BaseService):
    def list(self, *, query: dict[str, str] | None = None) -> dict[str, Any]:
        return self._request("GET", "/v1/environments", query=query)

    def create(self, body: Any) -> dict[str, Any]:
        return self._request("POST", "/v1/environments", body=body)

    def get(self, env_id: str) -> dict[str, Any]:
        return self._request(
            "GET", "/v1/environments/{envID}", path_params={"envID": env_id},
        )

    def update(self, env_id: str, body: Any) -> dict[str, Any]:
        return self._request(
            "PATCH", "/v1/environments/{envID}", path_params={"envID": env_id}, body=body,
        )

    def delete(self, env_id: str) -> dict[str, Any]:
        return self._request(
            "DELETE", "/v1/environments/{envID}", path_params={"envID": env_id},
        )

    def list_variables(self, env_id: str) -> dict[str, Any]:
        return self._request(
            "GET", "/v1/environments/{envID}/variables", path_params={"envID": env_id},
        )


class AsyncEnvironmentsService(AsyncBaseService):
    async def list(self, *, query: dict[str, str] | None = None) -> dict[str, Any]:
        return await self._request("GET", "/v1/environments", query=query)

    async def create(self, body: Any) -> dict[str, Any]:
        return await self._request("POST", "/v1/environments", body=body)

    async def get(self, env_id: str) -> dict[str, Any]:
        return await self._request(
            "GET", "/v1/environments/{envID}", path_params={"envID": env_id},
        )

    async def update(self, env_id: str, body: Any) -> dict[str, Any]:
        return await self._request(
            "PATCH", "/v1/environments/{envID}", path_params={"envID": env_id}, body=body,
        )

    async def delete(self, env_id: str) -> dict[str, Any]:
        return await self._request(
            "DELETE", "/v1/environments/{envID}", path_params={"envID": env_id},
        )

    async def list_variables(self, env_id: str) -> dict[str, Any]:
        return await self._request(
            "GET", "/v1/environments/{envID}/variables", path_params={"envID": env_id},
        )
